rest-api/src/main/java/com/cloudbees/workflow/rest/external/StatusExt.java

```diff
@@ -53,4 +57,22 @@ public static StatusExt valueOf(Throwable t) {
             return StatusExt.FAILED;
         }
     }
+
+    public static StatusExt fromGenericStatus(@CheckForNull GenericStatus st) {
+        if (st == null) {
+            return StatusExt.NOT_EXECUTED;
+        }
+        switch (st) {
+            case PAUSED_PENDING_INPUT: return StatusExt.PAUSED_PENDING_INPUT;
+            case ABORTED: return StatusExt.ABORTED;
+            case FAILURE: return StatusExt.FAILED;
+            case IN_PROGRESS: return StatusExt.IN_PROGRESS;
+            case UNSTABLE: return StatusExt.UNSTABLE;
+            case SUCCESS: return StatusExt.SUCCESS;
+            case NOT_EXECUTED: return StatusExt.NOT_EXECUTED;
+            default:
+                // Shouldn't happen, above includes all statuses
```

Comment:
Better to have IllegalStateException in order to propagate an error about the bug in the code if somebody adds new state and forgets about this code